packages/common/src/objects/deepFilter.ts

```diff
@@ -0,0 +1,39 @@
+import { isFindable } from "./internal/isFindable";
+import { setProp } from "./set-prop";
+
+/**
+ * Traverse a graph filtering out entries that do
+ * not pass the predicate
+ * @param object
+ * @param predicate
+ */
+export function deepFilter(
```

Comment:
Did you look at using generics so the typing can flow through? Something like `deepFilter<T>(object: T, predicate: (object: T) => boolean): T`

Reply:
I didn't look into this. It is intriguing, but I'm running into several other typing issues when I try to implement this. I'm going to get this merged in and will circle back to see if I can improve the typing using generics